Sources/FaunaDB/JSON.swift

```diff
@@ -197,6 +197,26 @@ private extension JsonType {
         }
     }
 
+    private func toRefV(object: [String: JsonType]) throws -> Value {
+        guard case let .string(id)? = object["id"] else { throw JsonError.invalidRefValue }
+
+        if object["class"] == nil && object["database"] == nil {
```

Comment:
I'd avoid the key lookup twice. You can do the same if statement agains the `clazz` and `database` variables below.